pmlc/dialect/comp/ir/interfaces.td

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 Intel Corporation
+#ifndef __PMLC_COMP_INTERFACES__
+#define __PMLC_COMP_INTERFACES__
+
+include "mlir/IR/OpBase.td"
+
+class COMP_OpInterface<string name> : OpInterface<name>;
+
+def COMP_ExecEnvOpInterface : COMP_OpInterface<"ExecEnvOpInterface"> {
+  let description = [{
+    Ties operation to specific execution environment.
+  }];
+
+  let methods = [
+    InterfaceMethod<"", "::mlir::Value", "getExecEnv", (ins), [{}], [{
+      ConcreteOp op = ::mlir::cast<ConcreteOp>(this->getOperation());
+      return op.execEnv();
+    }]>
+  ];
+}
+
+def COMP_ScheduleOpInterface : COMP_OpInterface<"ScheduleOpInterface"> {
```

Comment:
I'd probably call this AsyncOp or SchedulableOp; this interface isn't being used to perform scheduling, it's the thing being scheduled.

Reply:
I'm not sure if I understand this, all operations that implement this interface are expected to schedule work on device.
Also there is no differentiation between operations that are scheduled and that perform scheduling inside this dialect. Only operation I could see being "schedulable" is `gpu.launch_func`, but that's not a part of dialect.